### PR TITLE
The Mendix Portable Portal app

### PR DESCRIPTION
**Pull request by Luyao-Zhang-1**

### CODE REVIEW

File: content/en/docs/developerportal/mendix-portable-portal.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+After you install the Portable Portal app, open the app and sign in with your Mendix account. Then the [App Selection](#app-selection) page opens.
+
+## 3 App Selection Page {#app-selection}
+
+{{< figure src="/attachments/developerportal/mendix-portable-portal/app-selection.png" >}}
+
+The **App Selection** page shows all the apps that you can access in the Developer Portal. 
+
+Select apps that you want to manage in the Portal Portal app. You can select as many apps as needed. To select an app, tap the app. You can also search for an app using the search bar on the top.
```
Comment (MarkvanMents) [inline content/en/docs/developerportal/mendix-portable-portal.md]: You must select at least one app.

File: content/en/docs/developerportal/mendix-portable-portal.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+The **Community** page shows community content. The page has the following sections:
+
+* **Forum Questions** – shows the latest questions from the [Mendix Forum](/developerportal/community-tools/mendix-forum/)
+* **Events** – shows the upcoming Mendix events
+* **Blogs** – shows the latest community blog articles
+* **Mendix Youtube** – shows the Mendix Youtube videos
+
+In each section, you can swipe right or left to check out the next or previous item. Tapping **Show all** displays all the items in the section. If you tap an item, you will view its details in the web browser.
+
+## 6 Notifications Page
```
Comment (MarkvanMents) [inline content/en/docs/developerportal/mendix-portable-portal.md]: Do we need a list here of what notifications customers will receive?
And, more importantly, any notifications they won't receive?

I note that we class all notifications as Miscellaneous, so users can't turn off specific categories of notification. Might be nice to separate out the different types of notification so that a user can still receive cloud alerts without being overwhelmed with notifications about Buzz?

File: content/en/docs/developerportal/mendix-portable-portal.md [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+You can only manage licensed apps running on the Mendix Cloud. To do this, you will need to have the **Transport Rights** permission granted by the app's technical contact. For more information, see [Node Permissions](/developerportal/deploy/node-permissions/).
+
+Free apps will be shown with a red cross and a **Sandbox** environment indicator. Choosing a management option will return an error.
+
+Apps running in other clouds (for example, Mendix for Private Cloud) will display **No environments found**.
+{{% /alert %}}
+
+#### 4.1.1 Packages
+
+The **Packages** section will show you the latest package that has been created for your app. If you tap the package name, you will see details of the package.
```
Comment (MarkvanMents) [inline content/en/docs/developerportal/mendix-portable-portal.md]: What do the colours/symbols next to the package mean?

File: content/en/docs/developerportal/mendix-portable-portal.md [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+* Environment name – for example, Acceptance
+* Environment status – indicated by the color of the check mark next to the environment name:
+    * Green – there are no alerts
+    * Orange – there is at least one warning alert, but no critical alerts
+    * Red – there is at least one critical alert
+    
+    {{% todo %}}Is this information about caching also true for Portable Portal?{{% /todo %}}
+
+    ??The environment status is cached; there can be a delay of up to five minutes before the status icon displays a change of status. To see the details of the alerts, click **Alerts**. For more information, see [Alerts](/developerportal/operate/monitoring-application-health/).??
```
Comment (MarkvanMents) [inline content/en/docs/developerportal/mendix-portable-portal.md]: Is this caching information relevant to the portable portal?

File: content/en/docs/developerportal/mendix-portable-portal.md [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+    
+* Deployment package **Version** – for example, 1.0.0.18
+* **Runtime** – the Mendix Studio Pro version with which the app is built, for example, 9.5.0
+* The **URL** of the app – for example, `https://mytestapp.mendixcloud.com`
+
+Tapping the environment allows you to perform the following actions, if you have the **Transport Rights** permission for the environment:
```
Comment (MarkvanMents) [inline content/en/docs/developerportal/mendix-portable-portal.md]: These are all the actions I could see for my apps. Have I missed any?

**Review by MarkvanMents — COMMENTED** [review]
@Luyao-Zhang-1 
I've added the section on the Cloud Tab. I have added a couple of comments about things which I'm not sure about, but will watch the recording of the demo and see if I can pick anything else up.

Haven't included many screenshots - not sure how necessary they are - but can add more if you think it would improve things.

File: content/en/docs/developerportal/mendix-portable-portal.md [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+
+5. Scan the QR code to download and install the app.
+
+   {{% alert color="info" %}}If you are an iOS device user,  you are first directed to install TestFlight in App Store. After you install TestFlight, open it. Then you are prompted to install the Mendix Portable Portal app from TestFlight.{{% /alert %}}
+
+## 3 Starting the App
+
+After you install the Portable Portal app, open the app and sign in with your Mendix account. Then the [App Selection](#app-selection) page opens.
+
+## 3 App Selection Page {#app-selection}
```
Comment (MarkvanMents) [inline content/en/docs/developerportal/mendix-portable-portal.md]: Should this be section 4, or section 3.1?

File: content/en/docs/developerportal/mendix-portable-portal.md [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+
+{{< figure src="/attachments/developerportal/mendix-portable-portal/app-selection.png" >}}
+
+The **App Selection** page shows all the apps that you can access in the Developer Portal. 
+
+Select apps that you want to manage in the Portal Portal app. You can select as many apps as needed. To select an app, tap the app. You can also search for an app using the search bar on the top. 
+
+After you select the apps, tap **Continue**. Then the [My Apps](#my-apps) page opens.
+
+If this is the first time that you select apps on the **App Selection** page, you are prompted to select if you allow the Portable Portal app to send you notifications. We do recommend you select **Allow** since sending notifications is one of the main features of the app.
```
Comment (MarkvanMents) [inline content/en/docs/developerportal/mendix-portable-portal.md]: Would be nice if we could put this higher up, but I think it has to stay here to fit the app flow better?

File: content/en/docs/developerportal/mendix-portable-portal.md (hunk 2 above)
Comment (MarkvanMents) [reply content/en/docs/developerportal/mendix-portable-portal.md]: For example, we haven't mentioned that you only get notifications from cloud actions which are performed using the mobile app - not actions which are triggered from the developer portal (e.g. stopping the app in the Developer Portal will not trigger a notification in the mobile app).

Comment (MarkvanMents) [reply content/en/docs/developerportal/mendix-portable-portal.md]: And I wonder if the title should be **Notifications** rather than just the Notification Page because notifications appear in the phone UX (e.g. with icon badges and in the notification list) as well as within the app (depending on the Notification settings of the phone).

**Review by MarkvanMents — COMMENTED** [review]
@Luyao-Zhang-1 
I've added a couple more comments and updated the document based on the video of the demo.

File: content/en/docs/developerportal/mendix-portable-portal.md [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+You can only manage licensed apps running on the Mendix Cloud. To do this, you will need to have the **Transport Rights** permission granted by the app's technical contact. For more information, see [Node Permissions](/developerportal/deploy/node-permissions/).
+
+Free apps will be shown with a red cross and a **Sandbox** environment indicator. Choosing a management option will return an error.
+
+Apps running in other clouds (for example, Mendix for Private Cloud) will display **No environments found**.
+{{% /alert %}}
+
+#### 4.1.1 Packages
+
+The **Packages** section will show you the latest package that has been created for your app.
```
Comment (MarkvanMents) [inline content/en/docs/developerportal/mendix-portable-portal.md]: What do the colours/symbols next to the package mean?

_duplicating comment so it still appears here_

File: content/en/docs/developerportal/mendix-portable-portal.md (hunk 2 above)
Comment (Luyao-Zhang-1) [reply content/en/docs/developerportal/mendix-portable-portal.md]: > Do we need a list here of what notifications customers will receive? And, more importantly, any notifications they won't receive?
> 
> I note that we class all notifications as Miscellaneous, so users can't turn off specific categories of notification. Might be nice to separate out the different types of notification so that a user can still receive cloud alerts without being overwhelmed with notifications about Buzz?


@MarkvanMents The notifications that a user receive are (1) the cloud actions that are performed on the app and (2) the same notifications that they receive on the Developer Portal. The latter depends on user's [Notification Settings](https://user-settings.mendix.com/link/notifications). Because of this, maybe it's not easy to list a universal set of notifications that all users will (will not) receive?

File: content/en/docs/developerportal/mendix-portable-portal.md (hunk 7 above)
Comment (Luyao-Zhang-1) [reply content/en/docs/developerportal/mendix-portable-portal.md]: Indeed, this is the very moment when you are prompted to make the decision. Maybe it doesn't stand out enough? Should I turn this to a note?

File: content/en/docs/developerportal/mendix-portable-portal.md (hunk 2 above)
Comment (Luyao-Zhang-1) [reply content/en/docs/developerportal/mendix-portable-portal.md]: > And I wonder if the title should be **Notifications** rather than just the Notification Page because notifications appear in the phone UX (e.g. with icon badges and in the notification list) as well as within the app (depending on the Notification settings of the phone).

@MarkvanMents I would prefer to use **Notifications** page here, because this is about the UI inside the app, not the notifications that the user get in general on the phone. 

Even if the user doesn't allow the app to send push notification, on the **Notification** page inside the app, they can still receive notifications.

File: content/en/docs/developerportal/mendix-portable-portal.md (hunk 7 above)
Comment (MarkvanMents) [reply content/en/docs/developerportal/mendix-portable-portal.md]: Yes, an info box would be ideal!

File: content/en/docs/developerportal/mendix-portable-portal.md (hunk 2 above)
Comment (MarkvanMents) [reply content/en/docs/developerportal/mendix-portable-portal.md]: I think we should then add the following to this section:
1. A link to where the customer can decide which notifications they get from the platform (the one you provide above doesn't seem to be a valid link)
2. A note that they will receive IOS/Android notifications on their device, depending on their device settings.

**Review by MarkvanMents — COMMENTED** [review]
Hi @Luyao-Zhang-1 
I've made a couple more comments. We can address these together if you want, or you can just make some changes and resolve those comments.

The remaining questions are for the developers, so we can ask them to do a review.
Mark

File: content/en/docs/developerportal/mendix-portable-portal.md (hunk 2 above)
Comment (Luyao-Zhang-1) [reply content/en/docs/developerportal/mendix-portable-portal.md]: @MarkvanMents Thanks. I implemented point 2. 

Regarding point 1: Indeed, strange that the URL doesn't work, because if I click the notifications settings icon in the Developer Portal (please see the screenshot), I will be able to open the page with the same URL. I am checking with the dev why visiting the URL directly doesn't work. 

For now, the text there has a cross-reference to the [Notifications](https://docs.mendix.com/developerportal/#notifications) section in the _Developer Portal Guide_. The cross-referenced section contains info about how to configure the types of notifications that users can receive. (The link to the notifications settings page doesn't work here, either. I will fix that.) Hopefully that is enough?


image

**Comment by Luyao-Zhang-1** [comment]
> Hi @Luyao-Zhang-1 I've made a couple more comments. We can address these together if you want, or you can just make some changes and resolve those comments.
> 
> The remaining questions are for the developers, so we can ask them to do a review. Mark

Thanks Mark. I just asked Michael to help us review the doc.

File: content/en/docs/developerportal/mendix-portable-portal.md (hunk 2 above)
Comment (MarkvanMents) [reply content/en/docs/developerportal/mendix-portable-portal.md]: Great - thanks for doing that. You can resolve this now.

File: content/en/docs/developerportal/mendix-portable-portal.md (hunk 4 above)
Comment (MarkvanMents) [reply content/en/docs/developerportal/mendix-portable-portal.md]: Checked with #mobile-portal on Slack. Environment status is updated every few seconds.
This information is not relevant.

File: content/en/docs/developerportal/mendix-portable-portal.md [hunk 9]
```diff
@@ -3,158 +3,145 @@ title: "Mendix Portable Portal App"
...
 
 {{< figure src="/attachments/developerportal/mendix-portable-portal/app-selection.png" >}}
 
-The **App Selection** page shows all the apps that you can access in the Developer Portal. 
-
-Select at least one app that you want to manage in the Portal Portal app. You can select multiple apps. To select an app, tap the app. You can also search for an app using the search bar on the top. 
+Select at least one app that you want to manage in the Portable Portal app (you can also select multiple apps). To select an app, tap the app's name. You can also search for an app using the search bar on the top. 
 
-After you select the apps, tap **Continue**. Then the [My Apps](#my-apps) page opens.
+After you select the app(s), tap **Continue** to open the [My Apps](#my-apps) opens.
```
Comment (MarkvanMents) [inline content/en/docs/developerportal/mendix-portable-portal.md]: Typo.
```suggestion
After you select the app(s), tap **Continue** to open the [My Apps](#my-apps).
```

File: content/en/docs/developerportal/mendix-portable-portal.md [hunk 10]
```diff
@@ -3,158 +3,145 @@ title: "Mendix Portable Portal App"
...
-## 5 My Apps Page {#my-apps}
+## 4 My Apps {#my-apps}
 
-The **My Apps** page by default shows the last app that you selected from the [App Selection](#app-selection) page. The name of this app is shown on top of the screen. If you tap the app name, then you get the option to open a different app that you selected, or go back to the **App Selection** page where you can change the selection of the apps.
+The **My Apps** page by default shows the last app that you selected from the [App Selection](#app-selection) page:
 
 {{< figure src="/attachments/developerportal/mendix-portable-portal/my-apps.png" >}}
 
-The **My Apps** page contains two tabs: the **Cloud** tab and the **Stories** tab. The descriptions of these two tabs are in the following sections.
+The name of this app is shown on top of the screen. If you tap the app name, then you see the option to open a different app you previously selected or to go back to the **App Selection** page where you can change what apps you selected.
```
Comment (MarkvanMents) [inline content/en/docs/developerportal/mendix-portable-portal.md]: I prefer "at the top" when describing the screen, but it doesn't matter as long as we are consistent.
```suggestion
The name of this app is shown at the top of the screen. If you tap the app name, then you see the option to open a different app you previously selected or to go back to the **App Selection** page where you can change what apps you selected.
```

File: content/en/docs/developerportal/mendix-portable-portal.md [hunk 11]
```diff
@@ -3,158 +3,145 @@ title: "Mendix Portable Portal App"
...
 
-## 4 App Selection Page {#app-selection}
+The **App Selection** page shows all the apps that you can access in the Developer Portal:
 
 {{< figure src="/attachments/developerportal/mendix-portable-portal/app-selection.png" >}}
 
-The **App Selection** page shows all the apps that you can access in the Developer Portal. 
-
-Select at least one app that you want to manage in the Portal Portal app. You can select multiple apps. To select an app, tap the app. You can also search for an app using the search bar on the top. 
+Select at least one app that you want to manage in the Portable Portal app (you can also select multiple apps). To select an app, tap the app's name. You can also search for an app using the search bar on the top.
```
Comment (MarkvanMents) [inline content/en/docs/developerportal/mendix-portable-portal.md]: I prefer "at the top" when describing the screen, but it doesn't matter as long as we are consistent.
```suggestion
Select at least one app that you want to manage in the Portable Portal app (you can also select multiple apps). To select an app, tap the app's name. You can also search for an app using the search bar at the top. 
```

**Review by MarkvanMents — COMMENTED** [review]
@Adam-Dupaski 
This looks much better now with more active verbs.
Thanks.
I've made a couple of small suggestions, but none of them are necessary if you prefer the original.

File: content/en/docs/developerportal/mendix-portable-portal.md (hunk 10 above)
Comment (Adam-Dupaski) [reply content/en/docs/developerportal/mendix-portable-portal.md]: Sounds better

File: content/en/docs/developerportal/mendix-portable-portal.md (hunk 11 above)
Comment (Adam-Dupaski) [reply content/en/docs/developerportal/mendix-portable-portal.md]: Sounds better

**Comment by Adam-Dupaski** [comment]
@Luyao-Zhang-1 @MarkvanMents I think the quality of the screenshots can be improved, there's some blurriness and resolution issues there, etc. Maybe that can be done best after the release?

**Comment by Luyao-Zhang-1** [comment]
> @Luyao-Zhang-1 @MarkvanMents I think the quality of the screenshots can be improved, there's some blurriness and resolution issues there, etc. Maybe that can be done best after the release?

I will look into these issues. Do we already know the release date? @MarkvanMents

**Comment by MarkvanMents** [comment]
> > @Luyao-Zhang-1 @MarkvanMents I think the quality of the screenshots can be improved, there's some blurriness and resolution issues there, etc. Maybe that can be done best after the release?
> 
> I will look into these issues. Do we already know the release date? @MarkvanMents

Last contact with Jelena was in January - she said then that there was no fixed date then. @Luyao-Zhang-1 can you prompt her again to see if there is now a date for release?

**Comment by MarkvanMents** [comment]
Development work on this has halted and different solutions are being investigated.
Therefore closing this PR and deleting the branch.
Changes are retained offline in case they are needed in the future.
